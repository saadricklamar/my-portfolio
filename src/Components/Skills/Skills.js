import React from "react";
import "./Skills.scss";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faFilePdf } from "@fortawesome/free-solid-svg-icons";

function Skills() {
  return (
    <div className="Skills" id="mobile-skills">
      <main className="cards-container">
        <article className="skills-card">
          <section className="skills">
            <h5>JavaScript</h5>
            <section className="skill">
              <a
                href="https://reactjs.org/"
                target="_blank"
                rel="noopener noreferrer"
              >
                <img
                  src={require("../../assets/react.svg")}
                  alt="React logo"
                  width="50"
                  height="50"
                  className="logo"
                />
              </a>
              <p className="p-one">React</p>
            </section>
            <section className="skill">
              <a
                href="https://reactjs.org/"
                target="_blank"
                rel="noopener noreferrer"
              >
                <img
                  src={require("../../assets/redux.svg")}
                  alt="Redux logo"
                  width="50"
                  height="50"
                  className="logo"
                />
              </a>
              <p className="p-one">Redux</p>
            </section>
            <section className="skill">
              <a
                href="https://vuejs.org/"
                target="_blank"
                rel="noopener noreferrer"
              >
                <img
                  src={require("../../assets/vue.svg")}
                  alt="Kiwi standing on oval"
                  width="50"
                  height="50"
                  className="logo"
                />
              </a>
              <p className="p-one">Vue.js</p>
            </section>
            <section className="skill">
              <a
                href="https://jquery.com/"
                target="_blank"
                rel="noopener noreferrer"
              >
                <img
                  src={require("../../assets/jquery.svg")}
                  alt="Kiwi standing on oval"
                  width="50"
                  height="50"
                  className="logo"
                />
              </a>
              <p className="p-one">Jquery</p>
            </section>
          </section>
          <hr />
          <section className="skills">
            <h5>HTML/CSS</h5>
            <section className="skill">
              <a
                href="https://html.spec.whatwg.org/#is-this-html5?"
                target="_blank"
                rel="noopener noreferrer"
              >
                <img
                  src={require("../../assets/html5.svg")}
                  alt="HTML 5 logo"
                  width="45"
                  height="45"
                  className="logo"
                />
              </a>
              <p className="p-two">HTML5</p>
            </section>
            <section className="skill">
              <a
                href="https://developer.mozilla.org/en-US/docs/Archive/CSS3"
                target="_blank"
                rel="noopener noreferrer"
              >
                <img
                  src={require("../../assets/css3.svg")}
                  alt="CSS 3 logo"
                  width="45"
                  height="45"
                  className="logo"
                />
              </a>
              <p className="p-two">CSS3</p>
            </section>
            <section className="skill">
              <a
                href="https://getbootstrap.com/"
                target="_blank"
                rel="noopener noreferrer"
              >
                <img
                  src={require("../../assets/bootstrap.svg")}
                  alt="Bootstrap logo"
                  width="45"
                  height="45"
                  className="logo"
                />
              </a>
              <p className="p-two">Bootstrap</p>
            </section>
            <section className="skill">
              <a
                href="https://sass-lang.com/"
                target="_blank"
                rel="noopener noreferrer"
              >
                <img
                  src={require("../../assets/sass.svg")}
                  alt="Sass logo"
                  width="45"
                  height="45"
                  className="logo"
                />
              </a>
              <p className="p-two">Sass</p>
            </section>
          </section>
          <hr />
          <section className="skills">
            <h5>Native-Apps</h5>
            <section className="skill">
              <a
                href="https://kotlinlang.org/"
                target="_blank"
                rel="noopener noreferrer"
              >
                <img
                  src={require("../../assets/kotlin.svg")}
                  alt="Kotlin logo"
                  width="45"
                  height="45"
                  className="logo"
                />
              </a>
              <p className="p-four">Kotlin</p>
            </section>
            <section className="skill">
              <a
                href="https://www.nativescript.org/"
                target="_blank"
                rel="noopener noreferrer"
              >
                <img
                  src={require("../../assets/nativescript.svg")}
                  alt="NativeScript logo"
                  width="45"
                  height="45"
                  className="logo"
                />
              </a>
              <p className="p-four">NativeScript</p>
            </section>
            <section className="skill">
              <a
                href="https://facebook.github.io/react-native/"
                target="_blank"
                rel="noopener noreferrer"
              >
                <img
                  src={require("../../assets/react-native.svg")}
                  alt="React Native logo"
                  width="45"
                  height="45"
                  className="logo"
                  id="r-native"
                />
              </a>
              <p className="p-three" id="r-native-text">
                React Native
              </p>
            </section>
            <section className="skill">
              <a
                href="https://vue-native.io/"
                target="_blank"
                rel="noopener noreferrer"
              >
                <img
                  src={require("../../assets/vue.svg")}
                  alt="Vue Native logo"
                  width="45"
                  height="45"
                  className="logo"
                  id="v-native"
                />
              </a>
              <p className="p-three" id="v-native-text">
                Vue Native
              </p>
            </section>
          </section>
          <hr />
          <section className="skills">
            <h5 id="back-end">Back-End</h5>
            <section className="skill">
              <a
                href="https://nodejs.org/en/"
                target="_blank"
                rel="noopener noreferrer"
              >
                <img
                  src={require("../../assets/nodejs.svg")}
                  alt="Node JS logo"
                  width="45"
                  height="45"
                  className="logo"
                />
              </a>
              <p className="p-three">Node.js</p>
            </section>
            <section className="skill">
              <a
                href="https://www.postgresql.org/"
                target="_blank"
                rel="noopener noreferrer"
              >
                <img
                  src={require("../../assets/postgresql.svg")}
                  alt="Postgress logo"
                  width="45"
                  height="45"
                  className="logo"
                />
              </a>
              <p className="p-three">PostgreSQL</p>
            </section>
            <section className="skill">
              <a
                href="http://knexjs.org/"
                target="_blank"
                rel="noopener noreferrer"
              >
                <img
                  src={require("../../assets/knex.svg")}
                  alt="Knex logo"
                  width="45"
                  height="45"
                  className="logo"
                  id="knex"
                />
              </a>
              <p className="p-four" id="knex-text">
                Knex
              </p>
            </section>
            <section className="skill">
              <a
                href="https://www.getpostman.com/"
                target="_blank"
                rel="noopener noreferrer"
              >
                <img
                  src={require("../../assets/postman.svg")}
                  alt="Postman logo"
                  width="45"
                  height="45"
                  className="logo"
                  id="postman"
                />
              </a>
              <p className="p-four" id="postman-text">
                Postman
              </p>
            </section>
          </section>
          <hr />
          <section className="skills">
            <h5 id="back-end">Other</h5>
            <section className="skill">
              <a
                href="https://github.com/"
                target="_blank"
                rel="noopener noreferrer"
              >
                <img
                  src={require("../../assets/github.svg")}
                  alt="Github logo"
                  width="45"
                  height="45"
                  className="logo"
                />
              </a>
              <p className="p-five">Github</p>
            </section>
            <section className="skill">
              <a
                href="https://travis-ci.org/"
                target="_blank"
                rel="noopener noreferrer"
              >
                <img
                  src={require("../../assets/travis-ci.svg")}
                  alt="Travis CI logo"
                  width="45"
                  height="45"
                  className="logo"
                />
              </a>
              <p className="p-five" id="travis-text">
                Travis CI
              </p>
            </section>
            <section className="skill">
              <a
                href="https://mochajs.org/"
                target="_blank"
                rel="noopener noreferrer"
              >
                <img
                  src={require("../../assets/Jest.svg")}
                  alt="Jest logo"
                  width="45"
                  height="45"
                  className="logo"
                  id="jest"
                />
              </a>
              <p className="p-five">Jest</p>
            </section>
            <section className="skill">
              <a
                href="https://www.getpostman.com/"
                target="_blank"
                rel="noopener noreferrer"
              >
                <img
                  src={require("../../assets/mocha.svg")}
                  alt="Mocha logo"
                  width="45"
                  height="45"
                  className="logo"
                  id="postman"
                />
              </a>
              <p className="p-five" id="mocha-text">
                Mocha
              </p>
            </section>
          </section>
        </article>
      </main>
    </div>
  );
}

export default Skills;
