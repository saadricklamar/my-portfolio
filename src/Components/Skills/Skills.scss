// Skills Card //

.cards-container {
  display: flex;
  justify-content: left;
}

.skills-card {
  background-color: #e4eeef;
  border-radius: 1%;
  border: 1px solid #ccc;
  display: flex;
  flex-direction: column;
  height: 550px;
  margin: 8% 0 0 5%;
  overflow: scroll;
  padding: 0;
  width: 50%;

  &::-webkit-scrollbar-corner {
    background: #e4eeef;
  }
  &::-webkit-scrollbar {
    width: 5px;
    height: 5px;
  }

  &::-webkit-scrollbar-thumb {
    background-color: #17cddf;
    background-image: -webkit-gradient(
      linear,
      100% 0,
      0 0,
      color-stop(0.5, rgba(255, 255, 255, 0.2)),
      c color-stop(0.5, transparent),
      to(transparent)
    );
  }
}

.skills {
  display: flex;
}

// Each Skills Row //

.skill {
  display: flex;
  flex-direction: column;
  margin: 5%;
}

h5 {
  font-weight: bold;
  font-size: 1rem;
  margin: 10% 0 0 5%;
  padding: 0;
  text-decoration: underline;
}

.logo {
  margin-left: 10%;

  &:hover {
    cursor: pointer;
    filter: invert(0.4) saturate(20) hue-rotate(360deg) brightness(1.2);
  }
}

hr {
  display: block;
  height: 1px;
  border: 0;
  border-top: 1px solid #ccc;
  margin: 0;
  padding: 0;
}

// JavaScript Row //

.p-one {
  font-size: 1rem;
  margin: 0 10% 0 24%;
}

// HTML/CSS Row //

.p-two {
  font-size: 1rem;
  margin-bottom: 0;
  padding-left: 10%;
}

// Native-Apps Row //

#r-native {
  padding-right: 10%;
  position: relative;
  right: 30%;
}
#r-native-text {
  position: relative;
  right: 25%;
  width: 90px;
}

// Back-End Row //

#back-end {
  width: 90px;
}

// Other Row //

#travis-text {
  width: 90px;
}

// Rules that apply to many rows //

.p-three,
.p-four,
.p-five {
  font-size: 1rem;
  margin-bottom: 0;
}

#knex,
#postman,
#jest {
  position: relative;
  right: 30%;
}

#knex-text {
  position: relative;
  right: 10%;
}

#postman-text,
#mocha-text {
  position: relative;
  right: 20%;
}

// Resume Section //

.resume-section {
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin-left: 10%;
  margin-bottom: 10%;
}

.resume {
  background: none;
  border: none;
  color: black;
  font-family: "Titillium Web", sans-serif;
  font-size: 3rem;
  font-weight: bold;
  height: 0;
  outline: none;
  width: 260px;
}

.resume-icon {
  color: #f82a7b;
  font-size: 10rem;

  &:hover {
    color: #17cddf;
    cursor: pointer;
  }
}
