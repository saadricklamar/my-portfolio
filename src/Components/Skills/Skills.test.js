import React from "react";
import { shallow } from "enzyme";
import Skills from "./Skills";

describe("Skills", () => {
  let wrapper = shallow(<Skills/>);
  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});