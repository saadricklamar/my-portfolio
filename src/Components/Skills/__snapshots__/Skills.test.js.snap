// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skills should match the snapshot 1`] = `ShallowWrapper {}`;
